receiver/httpdreceiver/config_test.go

```diff
@@ -0,0 +1,213 @@
+			},
+		}
+		for _, tC := range testCases {
+			t.Run(tC.desc, func(t *testing.T) {
+				cfg := NewFactory().CreateDefaultConfig().(*Config)
+				cfg.Endpoint = tC.rawURL
+				require.NoError(t, cfg.Validate())
+				require.Equal(t, tC.expected, cfg.Endpoint)
+			})
+		}
+	})
+}
+
+func TestMissingProtocol(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		proto    string
+		expected bool
+	}{
+		{
+			desc:     "http proto",
+			proto:    "http://localhost",
+			expected: false,
+		},
+		{
+			desc:     "https proto",
+			proto:    "https://localhost",
+			expected: false,
+		},
+		{
+			desc:     "HTTP caps",
+			proto:    "HTTP://localhost",
+			expected: false,
+		},
+		{
+			desc:     "everything else",
+			proto:    "ht",
+			expected: true,
+		},
+		{
+			desc:     "everything else",
+			proto:    "localhost",
+			expected: true,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			require.Equal(t, tC.expected, missingProtocol(tC.proto))
+		})
+	}
+}
+
+func TestValidateEndpointFormat(t *testing.T) {
+	protocols := []string{"", "http://", "https://"}
+	hosts := []string{"", "127.0.0.1", "localhost", "customhost.com"}
+	ports := []string{"", ":8080", ":1234"}
+	paths := []string{"", "/server-status", "/custom"}
+	endpoints := []string{}
+	validEndpoints := map[string]bool{
+		"http://127.0.0.1:8080/server-status?auto":      true,
+		"http://127.0.0.1:1234/server-status?auto":      true,
+		"http://localhost:8080/server-status?auto":      true,
+		"http://localhost:1234/server-status?auto":      true,
+		"http://customhost.com:8080/server-status?auto": true,
+		"http://customhost.com:1234/server-status?auto": true,
+		"http://127.0.0.1:8080/custom?auto":             true,
+		"http://127.0.0.1:1234/custom?auto":             true,
+		"http://localhost:8080/custom?auto":             true,
+		"http://localhost:1234/custom?auto":             true,
+		"http://customhost.com:8080/custom?auto":        true,
+		"http://customhost.com:1234/custom?auto":        true,
+		// https
+		"https://127.0.0.1:8080/server-status?auto":      true,
+		"https://127.0.0.1:1234/server-status?auto":      true,
+		"https://localhost:8080/server-status?auto":      true,
+		"https://localhost:1234/server-status?auto":      true,
+		"https://customhost.com:8080/server-status?auto": true,
+		"https://customhost.com:1234/server-status?auto": true,
+		"https://127.0.0.1:8080/custom?auto":             true,
+		"https://127.0.0.1:1234/custom?auto":             true,
+		"https://localhost:8080/custom?auto":             true,
+		"https://localhost:1234/custom?auto":             true,
+		"https://customhost.com:8080/custom?auto":        true,
+		"https://customhost.com:1234/custom?auto":        true,
+	}
+
+	for _, protocol := range protocols {
+		for _, host := range hosts {
+			for _, port := range ports {
+				for _, path := range paths {
+					endpoint := fmt.Sprintf("%s%s%s%s?%s", protocol, host, port, path, "auto")
```

Comment:
```suggestion
					endpoint := fmt.Sprintf("%s%s%s%s?auto", protocol, host, port, path)
```
I'll make this change on this PR if there is any feedback, or otherwise on the next PR.